packages/nimble-components/src/rich-text-mention/base/models/mention-internals.ts

```diff
@@ -9,12 +11,18 @@ export interface MentionInternalsOptions {
 /**
  * Internal mention state
  */
-export class MentionInternals<TMentionConfig> {
+export class MentionInternals {
     /**
-     * Configuration which will hold mentioning info, character, icon and pattern
+     * Mention-specific configuration. Can be used to, for example, pass mention-specific configuration to views.
      */
     @observable
-    public mentionConfig?: TMentionConfig;
+    public mentionConfig?: MentionConfig;
```

Comment:
The intention of these mentionConfig objects was for state to be passed all the way to the view elements. I.e. from here all the way to the rich text mention view.
We haven't actually implemented that plumbing or test these objects. It also has a bit of a confusing name since we have some many objects that sound similar to "mention config".

To simplify and avoid having dead code can you remove these MentionConfig and UserMentionConfig classes as well as its associated createMentionConfig objects? If we end up needing that kind of configurability in the future we can reintroduce it.

Reply:
Removed `MentionConfig, UserMentionConfig and createMentionConfig()`